railties/lib/rails/test_unit/runner.rb

```diff
@@ -29,10 +29,8 @@ def parse_options(argv)
           $VERBOSE = argv.delete_at(w_index) if w_index
         end
 
-        def rake_run(argv = [])
-          ARGV.replace Shellwords.split(ENV["TESTOPTS"] || "")
-
-          run(argv)
+        def rake_test_opts
+          Shellwords.split(ENV["TESTOPTS"] || "")
```

Comment:
Can we do `require "fileutils"` and then do `FileUtils.sh` or just use the `system` method?

Reply:
We can't run FileUtils.sh because sh is only defined as an instance method: https://github.com/ruby/rake/blob/v13.0.3/lib/rake/file_utils.rb

We could be simply using `system` from TestRunner instead of `sh` from testing.rake. I thought using `sh` would bring us some more behavior, but double-checking, it's not especially using `verbose: false`.

Reply:
Cool, yeah then I'd just extend `rake_run` to shell out via `system` and add a comment as to why we need the shelling.

> We could be simply using `system` from TestRunner

I think `system` is from the `Kernel` module which is included in `Object`, which is how TestRunner responds to it.

Reply:
Yeah that makes sense, the diff is also much smaller. Another thing that `sh` was doing that I forgot to mention but that we have to do is to exit, in order to stop next tasks from running. Also `sh` was adding a backtrace that wasn't really useful, so `system` is much better. Added a test for that part.

> I think `system` is from the `Kernel` module which is included in `Object`, which is how TestRunner responds to it.

Yeah it's definitely Kernel#system, I just meant called from TestRunner.